vm/src/stdlib/math.rs

```diff
@@ -510,6 +510,36 @@ fn math_factorial(value: PyIntRef, vm: &VirtualMachine) -> PyResult<BigInt> {
     Ok(product)
 }
 
+fn math_perm(
+    n: PyIntRef,
+    k: OptionalArg<Option<PyIntRef>>,
+    vm: &VirtualMachine,
+) -> PyResult<BigInt> {
+    let n = n.as_bigint();
+    let v: BigInt = match k {
+        OptionalArg::Missing => n.clone(),
+        OptionalArg::Present(x) => match x {
+            Some(xi) => xi.as_bigint().clone(),
+            None => n.clone(),
+        },
+    };
```

Comment:
```suggestion
    let k_val;
    let v = match k.flatten() {
        Some(k) => {
            k_val = k.as_bigint().clone();
            &k_val
        }
        None => n,
    };
```
we can avoid a redundant clone of `k`